node/pkg/cni/token_watch.go

```diff
@@ -1,56 +1,158 @@
 package cni
 
 import (
+	"context"
 	"encoding/base64"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
+	"sync"
 	"time"
 
 	"github.com/sirupsen/logrus"
-	"gopkg.in/fsnotify/fsnotify.v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
-var serviceaccountDirectory string = "/var/run/secrets/kubernetes.io/serviceaccount/"
+const defaultCNITokenValiditySeconds = 3600
```

Comment:
This might be a little bit aggressive. Maybe I'm thinking about this wrong, but specifically for the large cluster scenario - 

Say you have 5000 nodes created at roughly the same time that all need to refresh their token - if you want to have a worse-case of about 5 seconds between each of those refresh API calls, that's 5000 * 5 = 25000 seconds to work with, or about 7 hours.

If we're using the jitter approach described in another comment below, that means you want your validity duration to be 3-4 times larger than that, so I'm thinking the validity duration should probably be a full day.

Reply:
Increased the default request validity period to a full day.